### ISSUE TITLE
Tracking Windows buildkit required features

### ISSUE THREAD

**Issue by junglie85** [adamrehn/ue4-docker#365]
Experimental Windows support was released in version 0.13.0 of buildkit and details of how to use are in a [blog](https://www.docker.com/blog/experimental-windows-containers-support-for-buildkit-released-in-v0-13-0/) entry.

This is a great step on the way to being able to unify Windows and Linux build processes.

What are the missing features in buildkit that are needed before unification can happen? Perhaps we could track them in this issue?

**Comment by slonopotamus** [comment]
I'm not sure we want to directly call buildkitd. Instead, it would be much more convenient if Windows + buildkit support was implemented for `docker buildx build`: https://github.com/moby/moby/blob/v27.3.1/builder/builder-next/executor_nolinux.go#L25

**Comment by TBBle** [comment]
Well, then we can mark down "integration into buildx" as an example of a feature we're waiting for. Off-hand, the only other thing that we care about that is new in buildx is the [`--secret` flag](https://docs.docker.com/reference/cli/docker/buildx/build/#secret), i.e. https://github.com/moby/buildkit/issues/5273 says that's not available yet, and AFAIR that's the primary motivator for using buildx/BuildKit.

**Comment by slonopotamus** [comment]
> the only other thing that we care about that is new in buildx is the [--secret flag](https://docs.docker.com/reference/cli/docker/buildx/build/#secret)

We also care about cache-mounts that only work with buildkit.

**Comment by TBBle** [comment]
Good point. As it happens, that's the same ticket, I suspect, in the general sense of "Get everything in https://github.com/moby/buildkit/blob/master/solver/llbsolver/mounts/mount.go working on WIndows". I'm not sure how cache mounts work at the implementation level, but all the other mounts in that file currently utilise tmpfs in some way or other, and we don't have a Windows equivalent for that AFAIK.

**Comment by slonopotamus** [comment]
I believe cache mounts are just binds.

**Comment by junglie85** [comment]
>I'm not sure we want to directly call buildkitd.
I'm hopeful that longer term this wouldn't be needed. But I don't see why ue4-docker would need to directly call buildkitd. Until buildkit is better integrated into Docker on Windows, starting containerd and buildkitd, and creating a builder could be a pre-requesite to running ue4-dockeer.

>the only other thing that we care about that is new in buildx is the [--secret flag](https://docs.docker.com/reference/cli/docker/buildx/build/#secret), i.e. https://github.com/moby/buildkit/issues/5273
This is probably a bigger blocker, given the statement that they don't plan to implement this until the classic builder has feature parity.

>I believe cache mounts are just binds.
I can't find confirmation of this in the docs. @slonopotamus do you know where this can be confirmed?

**Comment by profnandaa** [comment]
Hi folks -- I've got bind and cache mounts working and wanted your preliminary review on this as I complete out the rest of the stuff on the check-list -> https://github.com/moby/buildkit/pull/5708
/cc. @TBBle @slonopotamus

**Comment by slonopotamus** [comment]
https://github.com/moby/buildkit/issues/6567 was fixed
